src/transformers/models/jamba/modeling_jamba.py

```diff
@@ -709,11 +709,14 @@ def cuda_kernels_forward(
         # This is a hack to apply dt_proj while still using the forward pass of `torch.nn.Linear`, which is needed
         # in order to make quantization work. Quantization code replaces `torch.nn.Linear` layers with quantized
         # linear layers, and requires to call the forward pass directly.
-        # The original code here was: ```discrete_time_step = self.dt_proj.weight @ time_step.transpose(1, 2)```
-        time_proj_bias = self.dt_proj.bias
-        self.dt_proj.bias = None
+        # Quantized model can't work with the original code:
+        # ```discrete_time_step = self.dt_proj.weight @ time_step.transpose(1, 2)```
+        time_proj_bias = self.dt_proj.bias.data
+        with torch.no_grad():
+            self.dt_proj.bias.data = torch.zeros_like(self.dt_proj.bias.data)
         discrete_time_step = self.dt_proj(time_step).transpose(1, 2)
-        self.dt_proj.bias = time_proj_bias
+        with torch.no_grad():
+            self.dt_proj.bias.data = time_proj_bias
```

Comment:
Not sure why you would disable the gradients over here. The original code only seems to ignore the bias for the discrete time step.

Reply:
In general, I rather disable gradient computation wherever I can in order to save memory. Here we remove temporarily the bias, so the grads aren't needed.
This blocks wrapping aren't the main issue but the old None assignment that can't work in **full** training (comparing to lora/inference), I just added it as nice to have/save.